reana_server/status.py

```diff
@@ -251,11 +262,96 @@ def get_status(self):
         }
 
 
+class QuotaUsageStatus(REANAStatus):
+    """Class to retrieve statistics related to the current REANA users quota usage."""
+
+    def __init__(self, from_=None, until=None, user=None):
+        """Initialise QuotaUsageStatus class.
+
+        :param from_: From which moment in time to collect information. Not
+            implemented yet.
+        :param until: Until which moment in time to collect information. Not
+            implemented yet.
+        :param user: A REANA-DB user model.
+        :type from_: datetime
+        :type until: datetime
+        :type user: reana_db.models.User
+        """
+        super().__init__(from_=from_, until=until, user=user)
+
+    def get_top_five_all(self, resource_type):
+        """Get the top five users according to quota usage."""
+        subq = (
+            Session.query(UserResource)
+            .join(UserResource.resource)
+            .filter(Resource.type_ == resource_type)
+            .subquery()
+        )
+        subq_alias = aliased(UserResource, subq)
+        users = (
+            Session.query(User, subq_alias)
+            .join(subq_alias, UserResource)
+            .order_by(UserResource.quota_used.desc())
+            .limit(5)
+        )
+        return {
+            user.email: {
+                "id": str(user.id),
+                "used": ResourceUnit.human_readable_unit(
+                    resource.resource.unit, resource.quota_used
+                ),
+                "limit": ResourceUnit.human_readable_unit(
+                    resource.resource.unit, resource.quota_limit
+                ),
+            }
+            for user, resource in users
+        }
+
+    def get_top_five_limited(self, resource_type):
+        """Get the top five users according to quota usage per quota limit."""
+        subq = (
+            Session.query(UserResource)
+            .filter(UserResource.quota_limit != 0)
+            .join(UserResource.resource)
+            .filter(Resource.type_ == resource_type)
+            .subquery()
+        )
+        subq_alias = aliased(UserResource, subq)
+        users = (
+            Session.query(User, subq_alias)
+            .join(subq_alias, UserResource)
+            .order_by((UserResource.quota_used / UserResource.quota_limit).desc())
+            .limit(5)
+        )
```

Comment:
Minor comment: What do you think of extracting this logic to a private method?

Reply:
@diegodelemos I agree! I moved the query to a private method, what do you think?